### PR TITLE
Made PIDController more generic

### PR DESCRIPTION
**Pull request by anaselfathi**
I apologize for the double PR. your review wasn't lost since it was used in basal bolus therapy. 

For PID I changed the code to follow a more generic realization.
https://www.cds.caltech.edu/~murray/courses/cds101/fa02/caltech/astrom-ch6.pdf
http://www.cds.caltech.edu/~murray/books/AM08/pdf/am06-pid_16Sep06.pdf

### CODE REVIEW

**Comment by Gripnook** [comment]
Your PR conflicts with the main branch because you didn't pull the latest changes before making the updates.

You need to pull the latest changes, which includes PIDController, then apply your updates over that so the changes can be reviewed.

**Comment by anaselfathi** [comment]
Done ! again sorry about this !

File: library/controllers/@PIDController/PIDController.m [hunk 1]
```diff
@@ -74,11 +81,11 @@
             % Parse options.
             this.opt = struct();
             this.opt.name = this.name;
-            this.opt.glucoseTarget = 5.5; % mmol/L.
+            this.opt.glucoseTarget = 7.0; % mmol/L.
             this.opt.kP = 1e-2;
```
Comment (Gripnook) [inline library/controllers/@PIDController/PIDController.m]: This should also be `2e-2` to match the change above.

**Comment by anaselfathi** [comment]
Done

File: library/controllers/@PIDController/PIDController.m [hunk 2]
```diff
@@ -1,6 +1,10 @@
 classdef PIDController < InfusionController
     %PIDCONTROLLER  Proportional-Integral-Derivative controller.
-    %   For more details, see https://en.wikipedia.org/wiki/PID_controller.
+    %   For more details, see
+    % * https://en.wikipedia.org/wiki/PID_controller.
```
[the remark this answers is not in the document]
Comment (anaselfathi) [reply library/controllers/@PIDController/PIDController.m]: e75ccd33e0d39c2fabc1af6f981bcf6314a54a22

**Pull request merged**